### PR TITLE
Add option to provide custom user transformer

### PR DESCRIPTION
**Pull request by ianaktr**
This PR adds an option to provide a custom user transformer for okta users in user and org providers. It is a similar feature that LDAP has. If the transformer is provided it will be used instead of [userEntityFromOktaUser](https://github.com/RoadieHQ/roadie-backstage-plugins/blob/main/plugins/backend/catalog-backend-module-okta/src/providers/userEntityFromOktaUser.ts).

In our use case, we want to fill some of the fields of the user entity which isn't possible to do now. For example,  `profile.picture` or `metadata.description`.

Something similar was discussed in this issue https://github.com/RoadieHQ/roadie-backstage-plugins/issues/900

If this approach is accepted, I can also do that for groups.

#### :heavy_check_mark: Checklist

- [x] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [x] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by kissmikijr** [comment]
Fantastic thank you for this. It looks great. Would you mind adding to the groups as well?

**Comment by ianaktr** [comment]
Yes, of course! I will add this for groups as well 👍

**Comment by ianaktr** [comment]
@kissmikijr Hey! I added functionality to the groups as well and also updated the documentation. Would be nice if you could take a look once you have time. Thanks!

File: plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.test.ts [hunk 1]
```diff
@@ -422,5 +424,56 @@ describe('OktaGroupProvider', () => {
...
+      await provider.run();
+      expect(entityProviderConnection.applyMutation).toBeCalledWith({
+        type: 'full',
+        entities: expect.arrayContaining([
+          expect.objectContaining({
+            entity: expect.objectContaining({
+              kind: 'Group',
+              metadata: expect.objectContaining({
+                name: 'asdfwefwefwef',
+                description: 'Everyone in the company',
```
Comment (kissmikijr) [inline plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.test.ts:465]: Shouldn't we add the annotations to this as well?

File: plugins/backend/catalog-backend-module-okta/src/providers/types.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  Group as OktaGroup,
+  User as OktaUser,
+} from '@okta/okta-sdk-nodejs';
+import { GroupNamingStrategy } from './groupNamingStrategies';
+import { UserNamingStrategy } from './userNamingStrategies';
+
+export type OktaGroupEntityTransformer = (
+  group: OktaGroup,
+  namingStrategy: GroupNamingStrategy,
+  parentGroup: OktaGroup | undefined,
```
Comment (kissmikijr) [inline plugins/backend/catalog-backend-module-okta/src/providers/types.ts:28]: I think we could make parentGroup optional so we dont need to write out the explicit undefined here
```suggestion
  parentGroup?: OktaGroup,
```

**Review by kissmikijr — COMMENTED** [review]
Thank you! Left some tiny tiny comments, otherwise it looks good!

File: plugins/backend/catalog-backend-module-okta/src/providers/types.ts (hunk 2 above)
Comment (ianaktr) [reply plugins/backend/catalog-backend-module-okta/src/providers/types.ts]: Good catch! Done

File: plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.test.ts (hunk 1 above)
Comment (ianaktr) [reply plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.test.ts:465]: Sorry, I am not sure I understand. Do you mean to check annotations object in metadata? The untransformed group doesn't have any custom annotations that's why I didn't add any checks here. Only for the new `profile.displayName` which isn't available in the default transformer

**Review by kissmikijr — APPROVED** [review]
Thank you! Let's ship it!

**Pull request merged**